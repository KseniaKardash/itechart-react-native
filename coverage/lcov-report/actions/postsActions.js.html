<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for actions/postsActions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">actions</a> postsActions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.18% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.18% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/22</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* @flow */
import {
  ADD_POST,
  DELETE_POST,
  UPDATE_POST,
  GET_POSTS
} from "../constants/actionTypes";
import type {
  Post,
  Posts,
  AddPostAction,
  DeletePostAction,
  UpdatePostAction,
  GetPostsAction,
  FilterPostsAction
} from "../types/types";
import realm from "../database/index";
&nbsp;
export function updatePostAction(post: Post): UpdatePostAction {
  return {
    type: UPDATE_POST,
    post: post
  };
}
&nbsp;
export function deletePostAction(id: string): DeletePostAction {
  return {
    type: DELETE_POST,
    id: id
  };
}
&nbsp;
export function addPostAction(post: Post): AddPostAction {
  return {
    type: ADD_POST,
    post: post
  };
}
&nbsp;
export function getPostsAction(posts: Posts): GetPostsAction {
  return {
    type: GET_POSTS,
    posts: posts
  };
}
&nbsp;
type Dispatch = (action: Action | ThunkAction) =&gt; any;
type ThunkAction = (dispatch: Dispatch) =&gt; any;
type Action =
  | GetPostsAction
  | AddPostAction
  | DeletePostAction
  | UpdatePostAction
  | FilterPostsAction;
&nbsp;
export function <span class="fstat-no" title="function not covered" >addPost(</span>post: Post): ThunkAction {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >di</span>spatch =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    realm.write(<span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      realm.create("Post", post);</span>
    });
<span class="cstat-no" title="statement not covered" >    dispatch(addPostAction(post));</span>
  };
}
&nbsp;
export function <span class="fstat-no" title="function not covered" >updatePost(</span>post: Post): ThunkAction {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >di</span>spatch =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    realm.write(<span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      realm.create(</span>
        "Post",
        { id: post.id, description: post.description, tag: post.tag },
        true
      );
    });
<span class="cstat-no" title="statement not covered" >    dispatch(addPostAction(post));</span>
  };
}
&nbsp;
export function <span class="fstat-no" title="function not covered" >deletePost(</span>id: string): ThunkAction {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >di</span>spatch =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    realm.write(<span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
      const post = <span class="cstat-no" title="statement not covered" >realm.objects("Post").filtered(`id = "${id}"`);</span>
<span class="cstat-no" title="statement not covered" >      realm.delete(post);</span>
    });
<span class="cstat-no" title="statement not covered" >    dispatch(deletePostAction(id));</span>
  };
}
&nbsp;
export function <span class="fstat-no" title="function not covered" >getPosts(</span>): ThunkAction {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >di</span>spatch =&gt; {</span>
    const postsData = <span class="cstat-no" title="statement not covered" >realm.objects("Post");</span>
    const posts = <span class="cstat-no" title="statement not covered" >postsData.map(<span class="fstat-no" title="function not covered" >po</span>st =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      return { ...post };</span>
    });
<span class="cstat-no" title="statement not covered" >    dispatch(getPostsAction(posts));</span>
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Sep 26 2018 18:49:45 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
