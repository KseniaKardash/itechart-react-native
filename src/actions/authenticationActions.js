/* @flow */
import * as types from "../constants/actionTypes";
import * as flowType from "../types/types";

export function requestSignIn(): flowType.RequestSignIn {
  return {
    type: types.SIGN_IN_REQUEST
  };
}
export function requestSignInSuccess(
  user: flowType.User
): flowType.RequestSignInSuccess {
  return {
    type: types.SIGN_IN_SUCCESS,
    user: user
  };
}
export function requestSignInError(
  error: boolean
): flowType.RequestSignInError {
  return {
    type: types.SIGN_IN_FAILURE,
    error: error
  };
}
export function requestSignOut(): flowType.RequestSignOut {
  return {
    type: types.SIGN_OUT_REQUEST
  };
}
export function requestSignOutSuccess(
  userId: number
): flowType.RequestSignOutSuccess {
  return {
    type: types.SIGN_OUT_SUCCESS,
    userId: userId
  };
}
export function requestSignOutError(
  error: boolean
): flowType.RequestSignOutError {
  return {
    type: types.SIGN_OUT_FAILURE,
    error: error
  };
}

export function changeCurrentUser(
  user: flowType.User
): flowType.ChangeCurrentUser {
  return {
    type: types.CHANGE_CURRENT_USER,
    user: user
  };
}

export function deleteAuthorizedUser(userId: number) {
  return {
    type: types.DELETE_AUTHORIZED_USER,
    userId: userId
  };
}
