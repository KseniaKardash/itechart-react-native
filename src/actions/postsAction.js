/* @flow */
import { AsyncStorage } from "react-native";
import {
  ADD_POST,
  DELETE_POST,
  UPDATE_POST,
  GET_POSTS
} from "../constants/actionTypes";
import testPosts from "../constants/testPosts";
import type { Post } from "../types/types";

type AddPostAction = { type: "ADD_POST", post: Post };
type DeletePostAction = { type: "DELETE_POST" };
type UpdatePostAction = { type: "UPDATE_POST" };
type GetPostsAction = { type: "GET_POSTS", posts: Array<Post> };

type Action =
  | AddPostAction
  | DeletePostAction
  | UpdatePostAction
  | GetPostsAction;

type Dispatch = (action: Action | ThunkAction) => any;
type ThunkAction = (dispatch: Dispatch) => any;

function getIndex(data, id) {
  let clone = JSON.parse(JSON.stringify(data));
  return clone.findIndex(obj => parseInt(obj.id) === parseInt(id));
}

export function addPost(post: Post): ThunkAction {
  return dispatch => {
    AsyncStorage.getItem("data", (err, posts) => {
      posts = JSON.parse(posts);
      posts.unshift(post);
      AsyncStorage.setItem("data", JSON.stringify(posts), () => {
        dispatch({ type: ADD_POST, post: post });
      });
    });
  };
}
export function getPosts(): ThunkAction {
  return dispatch => {
    AsyncStorage.getItem("data", async (err, posts) => {
      if (posts == null) {
        await AsyncStorage.setItem("data", JSON.stringify(testPosts));
      }
      dispatch({ type: GET_POSTS, posts: JSON.parse(posts) });
    });
  };
}

export function updatePost(post: Post): ThunkAction {
  return dispatch => {
    AsyncStorage.getItem("data", (err, posts) => {
      posts = JSON.parse(posts);
      var index = getIndex(posts, post.id);
      if (index !== -1) {
        posts[index]["description"] = post.description;
        posts[index]["tag"] = post.tag;
      }
      AsyncStorage.setItem("data", JSON.stringify(posts), () => {
        dispatch({ type: UPDATE_POST, post: post });
      });
    });
  };
}

export function deletePost(id: number): ThunkAction {
  return dispatch => {
    AsyncStorage.getItem("data", (err, posts) => {
      posts = JSON.parse(posts);
      const index = getIndex(posts, id);
      if (index !== -1) posts.splice(index, 1);
      AsyncStorage.setItem("data", JSON.stringify(posts), () => {
        dispatch({ type: DELETE_POST, id: id });
      });
    });
  };
}
