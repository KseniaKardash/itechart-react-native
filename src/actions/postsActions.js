/* @flow */
import {
  ADD_POST,
  DELETE_POST,
  UPDATE_POST,
  GET_POSTS,
  SEARCH_USER_NAME
} from "../constants/actionTypes";
import type {
  Post,
  Posts,
  AddPostAction,
  DeletePostAction,
  UpdatePostAction,
  GetPostsAction,
  FilterPostsAction
} from "../types/types";
import realm from "../database/index";

export function updatePostAction(post: Post): UpdatePostAction {
  return {
    type: UPDATE_POST,
    post: post
  };
}

export function deletePostAction(id: string): DeletePostAction {
  return {
    type: DELETE_POST,
    id: id
  };
}

export function addPostAction(post: Post): AddPostAction {
  return {
    type: ADD_POST,
    post: post
  };
}

export function getPostsAction(posts: Posts): GetPostsAction {
  return {
    type: GET_POSTS,
    posts: posts
  };
}

export function filterPostsAction(posts: Posts): FilterPostsAction {
  return {
    type: SEARCH_USER_NAME,
    posts: posts
  };
}

type Dispatch = (action: Action | ThunkAction) => any;
type ThunkAction = (dispatch: Dispatch) => any;
type Action =
  | GetPostsAction
  | AddPostAction
  | DeletePostAction
  | UpdatePostAction
  | FilterPostsAction;

export function addPost(post: Post): ThunkAction {
  return dispatch => {
    realm.write(() => {
      realm.create("Post", post);
    });
    dispatch(addPostAction(post));
  };
}

export function updatePost(post: Post): ThunkAction {
  return dispatch => {
    realm.write(() => {
      realm.create(
        "Post",
        { id: post.id, description: post.description, tag: post.tag },
        true
      );
    });
    dispatch(addPostAction(post));
  };
}

export function deletePost(id: string): ThunkAction {
  return dispatch => {
    realm.write(() => {
      const post = realm.objects("Post").filtered(`id = "${id}"`);
      realm.delete(post);
    });
    dispatch(deletePostAction(id));
  };
}

export function getPosts(): ThunkAction {
  return dispatch => {
    const postsData = realm.objects("Post");
    const posts = postsData.map(post => {
      return { ...post };
    });
    dispatch(getPostsAction(posts));
  };
}

export function filterPostsByUserName(query: string): ThunkAction {
  return dispatch => {
    const postsData = realm
      .objects("Post")
      .filtered(`userName CONTAINS[c] "${query}"`);
    const posts = postsData.map(post => {
      return { ...post };
    });
    dispatch(filterPostsAction(posts));
  };
}
