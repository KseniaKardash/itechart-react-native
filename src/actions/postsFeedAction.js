/* @flow */
import {
  SEARCH_USER_NAME,
  SET_TOGGLE_SEARCH_BUTTON_STATUS
} from "../constants/actionTypes";

type SearchNameAction = { type: "SEARCH_USER_NAME", searchName: string };
type SearchToggleStatusAction = {
  type: "SET_TOGGLE_SEARCH_BUTTON_STATUS",
  toggleSearchStatus: boolean
};

export function changeSearchName(value: string): SearchNameAction {
  return {
    type: SEARCH_USER_NAME,
    searchName: value
  };
}

export function setToggleSearchStatus(
  value: boolean
): SearchToggleStatusAction {
  return {
    type: SET_TOGGLE_SEARCH_BUTTON_STATUS,
    toggleSearchStatus: value
  };
}
