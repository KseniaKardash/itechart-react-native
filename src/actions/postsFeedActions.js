/* @flow */
import {
  SEARCH_USER_NAME,
  SET_TOGGLE_SEARCH_BUTTON_STATUS
} from "../constants/actionTypes";
import type { SearchToggleStatusAction } from "../types/types";
import realm from "../database/index";

export function setToggleSearchStatus(
  value: boolean
): SearchToggleStatusAction {
  return {
    type: SET_TOGGLE_SEARCH_BUTTON_STATUS,
    toggleSearchStatus: value
  };
}

export function filterPostsByUserName(query: string): ThunkAction {
  return dispatch => {
    const postsData = realm
      .objects("Post")
      .filtered(`userName CONTAINS[c] "${query}"`);
    const posts = postsData.map(post => {
      return { ...post };
    });
    dispatch(filterPostsAction(posts));
  };
}

export function filterPostsAction(posts: Posts): FilterPostsAction {
  return {
    type: SEARCH_USER_NAME,
    posts: posts
  };
}
