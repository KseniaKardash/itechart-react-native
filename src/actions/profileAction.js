/* @flow */
import { SET_USERNAME, SET_USER_PHOTO } from "../constants/actionTypes";

type SetUserNameAction = { type: "SET_USERNAME", userName: string };
type SetUserPhotoAction = { type: "SET_USER_PHOTO", userPhoto: string };

export function setUserName(userName: string): SetUserNameAction {
  return {
    type: SET_USERNAME,
    userName: userName
  };
}

export function setUserPhoto(userPhoto: string): SetUserPhotoAction {
  return {
    type: SET_USER_PHOTO,
    userPhoto: userPhoto
  };
}
