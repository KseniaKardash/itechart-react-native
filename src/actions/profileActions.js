/* @flow */
import { SET_USERNAME, SET_USER_PHOTO } from "../constants/actionTypes";
import type { SetUserNameAction, SetUserPhotoAction } from "../types/types";

export function setUserName(userName: string): SetUserNameAction {
  return {
    type: SET_USERNAME,
    userName: userName
  };
}

export function setUserPhoto(userPhoto: string): SetUserPhotoAction {
  return {
    type: SET_USER_PHOTO,
    userPhoto: userPhoto
  };
}
