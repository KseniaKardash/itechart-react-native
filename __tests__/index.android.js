import "react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { deletePostAction } from "../src/actions/postsActions";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

beforeEach(() => {
  store.clearActions();
});

afterEach(() => {
  expect(store.getActions()).toMatchSnapshot();
});

test("Dispatch action", () => {
  const id = 56;
  store.dispatch(deletePostAction(id));
});
