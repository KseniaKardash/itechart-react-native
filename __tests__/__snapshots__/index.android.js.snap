// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispatch action 1`] = `
Array [
  Object {
    "id": 56,
    "type": "DELETE_POST",
  },
]
`;
